### ISSUE TITLE
The checkout actions v3 fails in set up job on GitHub hosted runner configured as service

### ISSUE THREAD

**Issue by flezaalv** [actions/checkout#1013]
We are following the step to configure the self hosted runner as service (https://docs.github.com/en/actions/hosting-your-own-runners/configuring-the-self-hosted-runner-application-as-a-service). but we get this errors in the Set up job for all workflows launched:

```
Current runner version: '2.299.1'
...
Prepare workflow directory
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v3' (SHA:93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8)
Warning: Failed to download action 'https://api.github.com/repos/actions/checkout/tarball/93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8'. Error: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
Warning: Back off 20.676 seconds before retry.
```

**Comment by jfelipe72** [comment]
I'm having the same problem over here. Sometimes it happens with other actions, after passing the checkout.

It started more or less when [this incident](https://www.githubstatus.com/incidents/74y4xlr09qll) was reported by GitHub.

**Comment by etaf** [comment]
> I'm having the same problem over here. Sometimes it happens with other actions, after passing the checkout.
> 
> It started more or less when [this incident](https://www.githubstatus.com/incidents/74y4xlr09qll) was reported by GitHub.
> 
> ![image](https://user-images.githubusercontent.com/49200056/202502976-470469a6-a7da-4039-b890-be41a8a73500.png)
> 
> ![image](https://user-images.githubusercontent.com/49200056/202503209-1da6fed4-bddf-4c8b-b824-2370fa4bfb83.png)

Hi, I got the same problem with you. And find the solution here: https://docs.github.com/en/actions/hosting-your-own-runners/using-a-proxy-server-with-self-hosted-runners

I settup the proxy in .env file then it works.

**Comment by jfelipe72** [comment]
> Hi, I got the same problem with you. And find the solution here: https://docs.github.com/en/actions/hosting-your-own-runners/using-a-proxy-server-with-self-hosted-runners
> 
> I settup the proxy in .env file then it works.

In my case it looks like it was a firewall turned on by the networking team.

We switched to a network that didn't have a firewall enabled and everything worked again.

Thank's for your time!
